File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let low = 0;
        let high = nums.length - 1;

        while (low < high) {
            let mid = Math.floor((high - low) / 2) + low;

            if (nums[mid] <= nums[high]) {
                // still in order, look for inflection point to left
                // include mid because it could be inflection point and min num
                high = mid;
            } else {
                // dont need to include mid when inflection is to right because min num is to right of inflection point
                low = mid + 1;
            }
        }

let mid = Math.floor((high - low) / 2) + low;
        return nums[mid];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = 1

        while right < len(numbers) and left < right:
            sum = numbers[left] + numbers[right]
            if sum == target:
                return [left + 1, right + 1]
            if sum < target:
                if right == len(numbers) - 1:
                    left += 1
                else: 
                    right += 1
            if sum > target:
                right -= 1
                left += 1
        return []